machine-learning/Dockerfile

```diff
@@ -45,9 +45,25 @@ COPY ./patches/* /tmp/
 RUN git apply /tmp/*.patch
 
 RUN /bin/sh ./dockerfiles/scripts/install_common_deps.sh
+
+ENV CCACHE_DIR="/ccache"
 # Note: the `parallel` setting uses a substantial amount of RAM
-RUN ./build.sh --allow_running_as_root --config Release --build_wheel --update --build --parallel 17 --cmake_extra_defines\
-    ONNXRUNTIME_VERSION="${ONNXRUNTIME_VERSION}" --skip_tests --use_rocm --rocm_home=/opt/rocm
+RUN --mount=type=cache,target=/ccache \
+    ./build.sh \
+    --allow_running_as_root \
+    --config Release \
+    --build_wheel \
+    --update \
+    --build \
+    --parallel 17 \
+    --cmake_extra_defines \
+    ONNXRUNTIME_VERSION="${ONNXRUNTIME_VERSION}" \
+    CMAKE_HIP_ARCHITECTURES="gfx900;gfx908;gfx90a;gfx1030;gfx1100;gfx1101;gfx1102;gfx940;gfx941;gfx942;gfx1200;gfx1201" \
```

Comment:
Can you add gfx906 to the supported architectures?

This will allow Vega 20/GCN 5.1 cards to work.

Reply:
@savely-krasovsky gfx900 has been deprecated since ROCm 5.7 and ROCm 6.4 still supports gfx906. ROCm 6.4 is still built against gfx906, but certain libraries like Tensible has been removed (though workarounds exist and will be out of scope).

 In the meantime, I will look into why the composable kernel build is failing and will propose a fix in a separate PR.

Reply:
Thanks and will do.